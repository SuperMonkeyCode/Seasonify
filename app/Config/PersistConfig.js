import { AsyncStorage } from 'react-native';

export default {
  active: true,
  reducerVersion: '0.4',
  storeConfig: {
    storage: AsyncStorage,
    // blacklist: ['login', 'search'], // reducer keys NOT stored to persistence
    whitelist: ['tvshows', 'friends'], // Optionally, just specify the keys you DO want stored to
    // transforms: [immutablePersistenceTransform]
  },
};
