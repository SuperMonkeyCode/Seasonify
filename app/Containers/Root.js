import React from 'react';
import PerfMonitor from 'react-native/Libraries/Performance/RCTRenderingPerf';
import { Container } from 'native-base';
import { connect } from 'react-redux';

import navigationLogging from 'Libs/Logging';
import PersistConfig from 'Config/PersistConfig';
import DebugConfig from 'Config/DebugConfig';
import DrawerNavigation from 'Navigation/DrawerNavigation';
import ui from 'State/uiState';
import ToastMessage from './ToastMessage';

const mapStateToProps = null; // state => ({});

const mapActionsToProps = {
  startup: ui.actions.startup,
};

export class Root extends React.Component {
  componentDidMount() {
    if (!PersistConfig.active) {
      this.props.startup();
    }
    if (DebugConfig.PerfMonitor) {
      PerfMonitor.toggle();
      setTimeout(() => {
        PerfMonitor.start();
        setTimeout(() => {
          PerfMonitor.stop();
        }, DebugConfig.PerfMonitorSettings.recordingDuration);
      }, DebugConfig.PerfMonitorSettings.waitBeforeStart);
    }
  }

  render() {
    return (
      <Container>
        <ToastMessage />
        <DrawerNavigation onNavigationStateChange={navigationLogging} />
      </Container>
    );
  }
}

export default connect(mapStateToProps, mapActionsToProps)(Root);
