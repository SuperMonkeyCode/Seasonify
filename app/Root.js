/* @flow */
import { Container } from 'native-base';
import { connect } from 'react-redux';
import PerfMonitor from 'react-native/Libraries/Performance/RCTRenderingPerf';
import React from 'react';

import { uiActions } from 'Store/uiStore';
import DebugConfig from 'Config/DebugConfig';
import DrawerNavigation from 'Navigation/DrawerNavigation';
import PersistConfig from 'Config/PersistConfig';
import ToastMessage from 'Components/ToastMessage';
import navigationLogging from 'Libs/Logging';

const mapStateToProps = null; // state => ({});

const mapActionsToProps = {
  startup: uiActions.startup,
};

type Props = {
  startup: Function,
};

export class Root extends React.Component<void, Props, void> {
  componentDidMount() {
    /* istanbul ignore next */
    if (!PersistConfig.active) {
      this.props.startup();
    }
    /* istanbul ignore next */
    if (DebugConfig.PerfMonitor) {
      PerfMonitor.toggle();
      /* eslint-disable no-console */
      console.log(
        `Starting logging in ${DebugConfig.PerfMonitorSettings.waitBeforeStart / 1000}s`,
      );
      setTimeout(() => {
        console.log(
          `Logging STARTING (ends in ${DebugConfig.PerfMonitorSettings.recordingDuration /
            1000}s)`,
        );
        /* eslint-enable no-console */
        PerfMonitor.start();
        setTimeout(() => {
          PerfMonitor.stop();
        }, DebugConfig.PerfMonitorSettings.recordingDuration);
      }, DebugConfig.PerfMonitorSettings.waitBeforeStart);
    }
  }

  render() {
    return (
      <Container>
        <DrawerNavigation onNavigationStateChange={navigationLogging} />
        <ToastMessage />
      </Container>
    );
  }
}

export default connect(mapStateToProps, mapActionsToProps)(Root);
