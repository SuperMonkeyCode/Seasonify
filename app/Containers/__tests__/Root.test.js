import React from 'react';
import PersistConfig from 'Config/PersistConfig';
import { Root } from '../Root';

function setup(specificProps = {}) {
  const props = {
    startup: jest.fn(),
    ...specificProps,
  };
  const component = shallow(<Root {...props} />);
  return {
    component,
    props,
  };
}

describe('Rendering when default state', () => {
  const { component } = setup();
  it('should match', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('componentDidMount & PersistConfig not active', () => {
  const { component, props } = setup();
  PersistConfig.active = false;
  component.instance().componentDidMount();

  it('should call startup match', () => {
    expect(props.startup).toBeCalled();
  });
});
