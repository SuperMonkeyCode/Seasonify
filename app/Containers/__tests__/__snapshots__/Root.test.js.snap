// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering when default state should match 1`] = `
<Styled(Container)>
  <Connect(ToastMessage) />
  <NavigationContainer
    onNavigationStateChange={[Function]}
  />
</Styled(Container)>
`;
